### PR TITLE
Use Host ID as Raft ID

### PR DESCRIPTION
**Pull request by kbr-scylla**
Thanks to #12250, Host IDs uniquely identify nodes. We can use them as Raft IDs which simplifies the code and makes reasoning about it easier, because Host IDs are always guaranteed to be present (while Raft IDs may be missing during upgrade).

Fixes: https://github.com/scylladb/scylladb/issues/12204

### CODE REVIEW

**Comment by kbr-scylla** [comment]
Note: https://github.com/scylladb/scylladb/pull/12084 isn't in `master` yet (only in `next`), so I based this PR on `next` and made `next` as the target so it doesn't show unrelated commits.

I will change the target to `master` once https://github.com/scylladb/scylladb/pull/12084 passes promotion.

File: service/storage_service.cc [hunk 1]
```diff
@@ -1702,16 +1689,15 @@ storage_service::prepare_replacement_info(std::unordered_set<gms::inet_address>
     }
 
     auto dc_rack = get_dc_rack_for(replace_address);
-    auto raft_id = get_raft_id_for(_gossiper, replace_address);
 
     auto replace_host_id = _gossiper.get_host_id(replace_address);
-    slogger.info("Host {}/{} is replacing {}/{} using a different address", _db.local().get_config().host_id, get_broadcast_address(), replace_host_id, replace_address);
+    slogger.info("Host {}/{} is replacing {}/{}", _db.local().get_config().host_id, get_broadcast_address(), replace_host_id, replace_address);
```
Comment (kbr-scylla) [inline service/storage_service.cc:1687]: Sneaked-in a fix for the message (https://github.com/scylladb/scylladb/pull/12250#discussion_r1045626954)

**Comment by avikivity** [comment]
Looks good, indeed a great simplification.

I double-checked that the raft id in the peers table isn't in any released version.

**Review by avikivity — APPROVED** [review]

**Review by kostja — APPROVED** [review]

File: gms/application_state.hh [hunk 2]
```diff
@@ -38,11 +38,6 @@ enum class application_state {
     IGNORE_MSB_BITS,
     CDC_GENERATION_ID,
     SNITCH_NAME,
-    // RAFT ID is a server identifier which is maintained
```
Comment (asias) [inline gms/application_state.hh:41]: Good. The RAFT_SERVER_ID is gone.

**Comment by asias** [comment]
> Thanks to #12084, Host IDs uniquely identify nodes. We can use them as Raft IDs which simplifies the code and makes reasoning about it easier, because Host IDs are always guaranteed to be present (while Raft IDs may be missing during upgrade).
> 
> Fixes: #12204

#12084 is the old PR. We need to ref the new one without the quarantine thing, no?

**Comment by asias** [comment]
One more thing, what happens if we have both the old and new node in the cluster. 

For example

n1, n2, n3
n4 replaces n3 with the same host id of n3 (n4  still runs the old code)
n4 is down
n5 replaces n4 with a diff host id of n3 (n5 runs the new code)

I think this is fine.

File: db/system_keyspace.cc [hunk 3]
```diff
@@ -355,7 +355,7 @@ schema_ptr system_keyspace::built_indexes() {
 }
 
 /*static*/ schema_ptr system_keyspace::peers() {
-    constexpr uint16_t schema_version_offset = 1; // raft_server_id
+    constexpr uint16_t schema_version_offset = 0;
```
[the remark this answers is not in the document]
Comment (kbr-scylla) [reply db/system_keyspace.cc:358]: I don't know. @kostja do you remember why this was necessary?

**Comment by kbr-scylla** [comment]
> https://github.com/scylladb/scylladb/pull/12084 is the old PR. We need to ref the new one without the quarantine thing, no?

Sorry, fixed.

**Comment by kbr-scylla** [comment]
@avikivity would you like to merge?

File: db/system_keyspace.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (kostja) [reply db/system_keyspace.cc:358]: I incremented this offset when added a new field to the table. Now that the patch is reverted, this increment should be reverted as well - end result, not change to schema id.

File: service/storage_service.cc (hunk 1 above)
Comment (bhalevy) [reply service/storage_service.cc:1687]: ack

**Pull request merged**